civis/tests/test_civis.py

```diff
@@ -0,0 +1,27 @@
+import civis
+from civis.compat import mock
+
+import pytest
+
+
+@pytest.mark.parametrize('schema_tablename', [
+    'foo.bar', '"foo".bar', 'foo."bar"', '"foo"."bar"'
+])
+def test_get_table_id(schema_tablename):
+    """Check that get_table_id handles quoted schema.tablename correctly."""
+    mock_client = mock.Mock()
+    mock_client.get_database_id.return_value = 123
+
+    mock_tables = mock.MagicMock()
+    mock_tables.__getitem__.side_effect = {0: mock.Mock()}.__getitem__
+
+    mock_client.tables.list.return_value = mock_tables
+
+    civis.APIClient.get_table_id(mock_client,
+                                 table=schema_tablename, database=123)
```

Comment:
@stephen-hoover The Python 3 builds have passed, but the Python 2.7 one failed on this line (with `TypeError: unbound method get_table_id() must be called with APIClient instance as first argument (got Mock instance instead)`). Would it fine to just `@pytest.mark.skip` this test for Python 2.7? If not, I don't see how we can have a mock client object without also mocking out `get_table_id`. Or am I missing something?

Reply:
If you want a mock client object, it's better to use the `civis.tests.mocks.create_client_mock` function to get one (that will handle autospeccing for you). In this case, you might actually want to create a real client object and then replace the functions you want to mock with `mock.Mock` objects. What do you think of trying:
```python
from civis.tests.mocks import TEST_SPEC
client = APIClient(local_api_spec=TEST_SPEC, api_key='none')
client.get_database_id = mock.Mock(return_value=123)
```
and so forth. If you want to argue that there's so much mocking going on that the test isn't useful, I'm open to that. This test seems nice to have, but maybe a separate test on `civis.io.split_schema_tablename` is all that's needed here.

Reply:
Ah, `TEST_SPEC` is what I didn't know about! The test has been updated, and the builds have passed.